sdk/eventhub/Azure.Messaging.EventHubs/src/Authorization/EventHubSharedKeyCredential.cs

```diff
@@ -0,0 +1,90 @@
+        /// </summary>
+        ///
+        /// <param name="sharedAccessKeyName">The name of the shared access key to be used for authorization, as reported by the Azure portal.</param>
+        /// <param name="sharedAccessKey">The value of the shared access key to be used for authorization, as reported by the Azure portal.</param>
+        ///
+        public EventHubSharedKeyCredential(string sharedAccessKeyName,
+                                           string sharedAccessKey)
+        {
+            Guard.ArgumentNotNullOrEmpty(nameof(sharedAccessKeyName), sharedAccessKeyName);
+            Guard.ArgumentNotNullOrEmpty(nameof(sharedAccessKey), sharedAccessKey);
+
+            SharedAccessKeyName = sharedAccessKeyName;
+            SharedAccessKey = sharedAccessKey;
+        }
+
+        /// <summary>
+        ///   Retrieves the token that represents the shared access signature credential, for
+        ///   use in authorization against an Event Hub.
+        /// </summary>
+        ///
+        /// <param name="scopes">The access scopes to request a token for.</param>
+        /// <param name="cancellationToken">The token used to request cancellation of the operation.</param>
+        ///
+        /// <returns>The token representating the shared access signature for this credential.</returns>
+        ///
+        public override AccessToken GetToken(string[] scopes, CancellationToken cancellationToken) => throw new InvalidOperationException(Resources.SharedKeyCredentialCannotGenerateTokens);
+
+        /// <summary>
+        ///   Retrieves the token that represents the shared access signature credential, for
+        ///   use in authorization against an Event Hub.
+        /// </summary>
+        ///
+        /// <param name="scopes">The access scopes to request a token for.</param>
+        /// <param name="cancellationToken">The token used to request cancellation of the operation.</param>
+        ///
+        /// <returns>The token representating the shared access signature for this credential.</returns>
+        ///
+        public override Task<AccessToken> GetTokenAsync(string[] scopes, CancellationToken cancellationToken) => throw new InvalidOperationException(Resources.SharedKeyCredentialCannotGenerateTokens);
+
+        /// <summary>
+        /// Coverts to shared access signature credential.
+        /// </summary>
+        ///
+        /// <param name="eventHubResource">The Event Hubs resource to which the token is intended to serve as authorization.</param>
+        /// <param name="signatureValidityDuration">The duration that the signature should be considered valid; if not specified, a default will be assumed.</param>
+        ///
+        /// <returns>A <see cref="SharedAccessSignatureCredential" /> based on the requested shared access key.</returns>
+        ///
+        internal SharedAccessSignatureCredential CovertToSharedAccessSignatureCredential(string eventHubResource,
```

Comment:
Covert -> Convert?

Reply:
Spelling is hard...    :P

Reply:
Especially when you are part of a covert operation ....

Reply:
![](https://media1.tenor.com/images/2491c66823e1818f350535db87d2bde5/tenor.gif?itemid=5797030)